src/Mod/PartDesign/App/FeaturePad.cpp

```diff
@@ -71,11 +71,10 @@ App::DocumentObjectExecReturn *Pad::execute()
 {
     // Validate parameters
     double L = Length.getValue();
-    if ((std::string(Type.getValueAsString()) == "Length") && (L < Precision::Confusion()))
-        return new App::DocumentObjectExecReturn(QT_TRANSLATE_NOOP("Exception", "Length of pad too small"));
     double L2 = Length2.getValue();
-    if ((std::string(Type.getValueAsString()) == "TwoLengths") && (L < Precision::Confusion()))

```

Comment:
in any case I think here we want to check `L+L2 < Precision::Confusion()`

Reply:
it already fails here:
https://github.com/FreeCAD/FreeCAD/blob/bf8c59ba2ab4a2e6bd2060544130a0a4693ba4d8/src/Mod/PartDesign/App/FeatureExtrude.cpp#LL168C1-L174C1

Reply:
I that case I think we should just remove this check, IMO abs(L2)>0 and L==0 is a valid input

Reply:
And L2 < L1 < 0 should be valid too.

Reply:
I think L1 is restricted to positive values but not by checks

Reply:
You're right, see f9d153818 which remove the positive constraint for L2.
Negative L1 is similar to reverse mode but it could be helpful when work with expressions.

Reply:
I think it's a little too redundant with the "Reverse" option, would like to hear more opinions on the matter, if you think we should enable this please create an issue and let's discuss there